### ISSUE TITLE
Timesheets feature request: click to enter

### ISSUE THREAD

**Issue by gcapes** [UoMResearchIT/CapX#623]
To enter values in a category, you have to first click on the edit button, then the cell you want to edit. If you could just click on the cell to edit, and press enter to save, that would save some clicks/lost data/sanity

**Comment by ajerrison-manchester** [comment]
I've got a feeling that this is a feature of the Blazor component being used as all the other data entry screens in CapX are the same, however I agree it is not intuitive and would be nice to change. The first time I use any new screen in CapX I always make the same mistake. :-)

**Comment by aharwood2** [comment]
Yes, CapX uses the Radzen library of frontend components. That said, I'm certain we can either modify the component by extending so worth looking into.

**Comment by gcapes** [comment]
I guess it particularly gets me because the focus is "task based" rather than "day based". If I had to select a day to edit instead of a task row, that would also make the whole thing easier to fill out.

**Comment by gcapes** [comment]
Actually given this is a request, I'd like to request that values are saved on entry, without having to press enter or anything else.

**Comment by ianhinder** [comment]
I also find it annoying (in general in CapX) that you see a row of data in a table, and instead of just editing it in place by clicking on it, you have to find the Edit button, which gives you a slightly different UI, and then allows you to edit the field, and then have to confirm it. I think a more modern style would be to just allow editing in place. In some sense this was easier in Innate.... <ducks...>

**Comment by aharwood2** [comment]
I know what you mean and I do agree to a certain extent. There are lots of places in CapX where we use this [RadzenDataGrid](https://blazor.radzen.com/datagrid-inline-edit?theme=material3) component to both display records from a DB table and provide a mechanism for editing the records in the table without having to provide a separate edit page. This is usually the case for simpler models. One option could be to allow multiple entry editing and then just leave every row in "Edit Mode" by default, saving the changes in one go with a separate save button. Alternatively we can just use some kind of bootstrap solution or a different component. There are certainly options.

**Comment by aharwood2** [comment]
This is a new feature by the looks of it: https://blazor.radzen.com/datagrid-incell-edit?theme=material3

**Comment by ianhinder** [comment]
Entering my timesheets I've noticed how much time I spend moving the mouse to the Edit button and then back again, and the number of extra clicks I have to make to get each row updated. Having **in-place editing** would be a big improvement in ergonomics. As @carriking mentioned in #783, friction like this adds to the "dread [of] having to fill in the timesheet", and it seems like it would be an easy fix, especially if the component now supports it.

**Automatically saving** without having to click would also add to the user experience.

**Comment by ajerrison-manchester** [comment]
I am not fully up to speed on Blazor server-side yet, but I suspect automatic saving would involve some kind of client side message to say that editing was taking place (i.e. things like an 'on key up' message), which I presume server-side software wouldn't know about? Might be misunderstanding either your issue or the platform though. Or both. :-)

**Comment by ianhinder** [comment]
It already seems to do this for the Notes field at the bottom of the timesheet; after adding a note, when focus leaves the box, I see a little green box saying that it has been saved, which is nice.

**Comment by ajerrison-manchester** [comment]
Ah yes, looking at the code there is a trigger function on the 'Changed' event of the textarea called NoteChanged(). Basically the same idea as adding client-side JS in Webforms (no comment about whether we should be mixing function, structure and layout in the same page :-) ).

The Numeric boxes also have Changed event functions, but they don't called the DB update. I suspect because it would hammer the database and might interfere with the SaveRow function.

**Comment by ianhinder** [comment]
It's amazing to me that with all the computing and network power at our disposal, sending a couple of numbers ends up "hammering the database"! Not disagreeing; I don't know much about it - just complaining about bloat!

**Comment by ajerrison-manchester** [comment]
As far as I am aware it still uses SQLite - i.e. 1 write connection at a time for the whole database!

**Comment by ianhinder** [comment]
Yeah, but we have something like 14 users, not 140,000, and they're probably not all using it at the same time!

**Comment by ajerrison-manchester** [comment]
Believe it or not we have over 50 users at the moment, although a handful are not active. And entering values in your timesheet is probably something most people do at about the same time every day,

**Comment by ianhinder** [comment]
Ah, I'd forgotten of course that it's not just the G7s using CapX now.

**Comment by gcapes** [comment]
C'mon @ianhinder  - remember who created this issue in the first place 😄

**Comment by ianhinder** [comment]
I barely remember what I had for breakfast... And I'm unlikely to reread the whole thread when I comment ;')

**Comment by fherreazcue** [comment]
Just adding a ping on this issue, as it seems to have fallen through the cracks...

**Comment by aharwood2** [comment]
> Just adding a ping on this issue, as it seems to have fallen through the cracks...

Thanks, Francisco. It hasn't fallen through the cracks as such, just not planned for a particular release as things have been prioritised above it. I'll drop it into one of the upcoming releases and we can see if we get round to looking at it. We are slowly building the tooling working group with a few .NET folk who have time to work on the tool.
